var/spack/repos/builtin/packages/msmpi/package.py

```diff
@@ -31,9 +31,9 @@ class Msmpi(Package):
 
     @classmethod
     def determine_version(cls, exe):
-        output = Executable(exe)()
-        ver_str = re.search("[Version ([0-9.]+)]", output)
-        return Version(ver_str.group(0)) if ver_str else None
+        output = Executable(exe)(output=str, error=str)
+        ver_str = re.search(r"\[Version ([0-9.]+)\]", output)
+        return Version(ver_str.group(1)) if ver_str else None
```

Comment:
I don't *think* this would run into the same problem as https://github.com/spack/spack/pull/34574, but if we don't anticipate MSMPI installations on non-windows systems, then we may want to either

* Refine the regular expression to be less generic
* Always return `None` on non-Windows for this package

Reply:
> I'd rather keep the regex generic to support not specifying an extension for cross platform packages, 

To be clear, I was not referring to the regex used to identify the executable, but rather the regex used to extract the version. For example, right now something like `"Version 4.5"` would match, and it seems like other MPI implementations could easily have this string. If `msmpi` outputs something like `"Microsoft MPI Version 4.5"` then we could add that static string to the regex used in this function.

If `msmpi` could be installed on Linux, then I think refining the regex in this function is preferable. If it can't, then returning `None` on non-Windows would be fine.

Reply:
I've updated to return none. MSMPI is intended to be a Windows implementation of mpich, so even if it could be installed, mpich would likely be preferable, to the point where I think returning none is fine.

I misunderstood, thanks for clarifying. That regex actually does need updating, I intended for there to be two literal brackets there that I forgot to include when I refactored to have that regex actually capture what we want from it. I'll add the msmpi string as well.